[Term Entry] SQL NULL SAFE EQUAL Operator (#5205)
Codecademy/docs: content/sql/concepts/operators/terms/null-safe-equal/null-safe-equal.md

---
Title: 'NULL SAFE EQUAL'

> Description: 'Performs an inequality comparison similar to the '=' operator, using the logic of an XNOR gate. Can accept NULL as a value and will not output NULL.'

Reviewer: ```suggestion
Description: 'Performs an equality comparison similar to the '=' operator, using the logic of an XNOR gate. Can accept NULL as a value and ensures comparisons involving NULL do not result in NULL.
```
Here I have corrected  `inequality comparison` to `equality comparison` because that is the correct logic for null safe equal operator in SQL.

Subjects: 
  - 'Computer Science'

> Tags: 
>   - 'Operators'
>   - 'Database'
>   - 'Queries'
>   - 'Tables'

Reviewer: ```suggestion
Tags: 
  - 'Operators'
  - 'Database'
  - 'Queries'
  - 'Tables'
```
Include only those tags that are already present in the [tags.md list ](documentation/tags.md)and you can add one if you want to include it in your PR. Here `operators` is not part of the tags list so you can add it if you want it to be used.

CatalogContent: 
  - 'learn-sql'
  - 'paths/analyze-data-with-sql'
---

> The **'NULL SAFE EQUAL'** operator performs an equality comparison like the '=' operator. However it can also accept NULL values and never returns NULL, instead always returning TRUE or FALSE. It uses the logic of an XNOR gate to determine the value returned.

Reviewer: ```suggestion
The **`NULL SAFE EQUAL`** operator performs an equality comparison like the '=' operator. However, it can also accept NULL values and never returns NULL, instead always returning TRUE or FALSE. It uses the logic of an XNOR gate to determine the value returned.
```

> The 'NULL SAFE EQUAL' function is useful for when an XNOR gate that can accept NULL values and always returns True or False is needed.

Reviewer: ```suggestion
The `NULL SAFE EQUAL` operator is useful when an XNOR-like logic is required for comparisons that can accept NULL values and always return True or False.
```
The NULL SAFE EQUAL is an operator so it must be enclosed in `.....` backticks, also removed the function word as it is not a function.
Also, NULL SAFE EQUAL operator is useful when XNOR-like logic is required not literally an XNOR gate.

> **Note:** The syntax can vary across different SQL flavors. In PostgreSQL, the operator is written as 'a IS NOT DISTINCT FROM b'. In MariaDB/MySQL, it is written as 'a <=> b'. Finally, in SQLite, it is written as 'a IS b'.

> ## MySQL Syntax

Reviewer: ```suggestion
## Syntax
```
Removed MYSQL as it is not necessary here, only the word syntax is sufficient.

---
Title: 'NULL SAFE EQUAL'
Description: 'Performs an equality comparison similar to the '=' operator, using the logic of an XNOR gate. Can accept NULL and ensures comparisons involving NULL do not result in NULL.'
Subjects: 
  - 'Computer Science'
Tags:
  - 'Operators' 
  - 'Database'
  - 'Queries'
  - 'Tables'
CatalogContent: 
  - 'learn-sql'
  - 'paths/analyze-data-with-sql'
---

> The **`NULL SAFE EQUAL`** operator performs an equality comparison like the '=' operator. However, it can also accept NULL values and never returns NULL, instead always returning TRUE or FALSE. It uses the logic of an XNOR gate to determine the value returned.

Reviewer: ```suggestion
The **`NULL SAFE EQUAL`** operator performs an equality comparison like the '=' operator. However, it can also accept NULL values and never returns NULL; instead, it always returns TRUE or FALSE. It uses the logic of an XNOR gate to determine the value returned.
```

> The 'NULL SAFE EQUAL' operator is useful for when XNOR-like logic is required for comparisons that can accept NULL values and always return True or False is needed.

Reviewer: ```suggestion
The `NULL SAFE EQUAL` operator is useful when XNOR-like logic is required for comparisons that can accept NULL values and always return True or False.
```
NULL SAFE EQUAL is an operator therefore it must be enclosed in (`.....`) backticks.

> The `NULL SAFE EQUAL` operator is useful for when XNOR-like logic is required for comparisons that can accept NULL values and always return True or False is needed.

Reviewer: ```suggestion
The `NULL SAFE EQUAL` operator is useful when XNOR-like logic is required for comparisons that can accept NULL values and always return True or False.
```

> **Note:** The syntax can vary across different SQL flavors. In PostgreSQL, the operator is written as 'a IS NOT DISTINCT FROM b'. In MariaDB/MySQL, it is written as 'a <=> b'. Finally, in SQLite, it is written as 'a IS b'.

## Syntax

> ```pseudo
> SELECT column1
> FROM table
> WHERE column1 <=> condition
> ```

Reviewer: ```suggestion
```pseudo
SELECT column1
FROM table
WHERE column1 <=> value;
```
```
The NULL SAFE EQUAL operator <=> does the comparison between two values , not with the word "condition".
